firebase-installations/customer-lint-checks/src/main/java/com/google/firebase/installations/lint/IncompatibleIidVersionDetector.kt

```diff
@@ -68,18 +81,45 @@ class IncompatibleIidVersionDetector : Detector() {
         }
     }
 
+    private fun getVariants(project: Project): List<Variant> {
+        if (!project.isGradleProject) {
+            return listOf()
+        }
+
+        // using reflection here due to breaking change in lint-api 26.6.0 that changed the return type
+        // of getGradleProject()
+        val method = project.javaClass.getMethod("getGradleProjectModel")
+        method.isAccessible = true
+        val model = method.invoke(project)
+        val variantsMethod = model.javaClass.getMethod("getVariants")
+        variantsMethod.isAccessible = true
+        return variantsMethod.invoke(model) as List<Variant>
+    }
+
     private fun isCompatibleVersion(coordinates: MavenCoordinates): Boolean {
         val versionComponents = coordinates.version.split('.', limit = 3).toTypedArray()
 
         // Incompatible if major version is before v20
-        if (20 > versionComponents.get(0).toInt()) {
+        if (20 > versionComponents[0].toInt()) {
             return false
         }
         // Compatible if major version is after v21
-        if (21 <= versionComponents.get(0).toInt()) {
+        if (21 <= versionComponents[0].toInt()) {
```

Comment:
Can this be simplified? 
`return (versionComponents[0].toInt() >= 20 ) && (versionComponents[1].toInt() >= 1)`

Reply:
nvm. brain slip